Makefile

```diff
@@ -124,8 +127,21 @@ docs: typedoc generated-documentation
 view-docs:
 	PYTHONPATH=${ROOTDIR} PIPENV_IGNORE_VIRTUALENVS=1 expect -c 'spawn pipenv run mkdocs serve; expect "Serving on"; exec open "http://localhost:8000"; interact'
 
+###############################################################################
+### Deployment of documentation ### 
+
+# This step generates all the documentation for the SDK using mkdocs and dumps the contents in /site
+.PHONY: build-docs
+build-docs:
+	${PIPENV} run mkdocs build;
+
 .PHONY: publish-docs
-publish-docs:
+publish-docs: docs build-docs
+# TODO(spencer): change GIT SHA to the version from package.json
+	(cd ${ROOTDIR}; tsnode documentation/documentation_publisher.ts push ${CURRENT_GIT_SHA1})
```

Comment:
The binary is `ts-node`. Also this will only work if node_modules/.bin is on the path, so unless something is setting the path, probably want to reference it as node_modules/.bin/ts-node